### ISSUE TITLE
Mask with arrays crashes in React when adding value in the middle

### ISSUE THREAD

**Issue by homer0** [uNmAnNeR/imaskjs#647]
**Describe the bug**

Before anything, thanks for the amazing lib :D!

I have a simple mask for pixel units `num px`:

```js
{
  mask: [
    {
      mask: '',
    },
    {
      mask: 'num px',
      lazy: false,
      blocks: {
        num: {
          mask: Number,
          signed: false,
          scale: 0,
        },
      },
    },
  ],
};
```

Everything works great if you add numbers at the beginning of the value, or after the `px` (they get moved :)), but whenever I try to add a number in the middle, like in `20`, adding a `1` between `2` and `0`, it crashes.

**To Reproduce**

https://codesandbox.io/s/patient-morning-kcwibd

**Expected behavior**
Don't crash

**Environment:**
 - OS: MacOs
 - Browser Edge
 - Version 99.0.1150.52
 - IMask version 6.4.2
 - Framework/plugin version if used: `react-imask`

**Additional context**

The problem seems to be [here](https://github.com/uNmAnNeR/imaskjs/blob/master/packages/imask/src/masked/pattern.js#L288), when `flags._beforeTailState` is not `undefined`, but `blocks` is; sorry, I don't have the enough context to send a PR.

**Comment by alvesvin** [comment]
Same here. What's funny is that `blocks` in the source code has the ? operator, but the compiled ESM doesn't include any undefined/null check for `blocks` only for `flags._beforeTailState`.

**Comment by alvesvin** [comment]
Oh I see. It's recent change [here](https://github.com/uNmAnNeR/imaskjs/commit/6f82842a0feebf8359d63fb5a578336883171a9e#diff-cbe77583d243880b0cd5777def1ef550c9604e70c85f2d99451d813806c7fff2). Seems to be fixed in #701

**Comment by uNmAnNeR** [comment]
fixed in #701

**Comment by cmocanu** [comment]
@uNmAnNeR it doesn't seem to be released, yet, right? Latest seems to be `6.4.2`, checking both npm and the tags in the repo. I'm dealing with this same issue, any estimate when there'll be a new release?

**Comment by uNmAnNeR** [comment]
@cmocanu very soon
